app/services/scholars_archive/caching_service.rb

```diff
@@ -0,0 +1,21 @@
+require 'net/http'
+
+module ScholarsArchive
+  class CachingService
+    def self.fetch_or_store_in_cache(vocabulary)
+      Rails.cache.fetch("/caching/#{vocabulary}", expires_in: 1.day, race_condition_ttl: 15.seconds) do
```

Comment:
I believe `#fetch` is taking a string as the key.. if you verify that, then the key should just be the argument `uri` .. that way there isn't an unintended implied relationship between this key and the `config.cache_store`